### ISSUE TITLE
Uncertainties for ternary+ mixtures

### ISSUE THREAD

**Issue by Akwiatkowski-eng** [usnistgov/REFPROP-issues#706]
### Description
Hello, 
I'm not sure if that is the right place to ask this question. If it is not, please direct me.

I'm working on a paper in which I need to specify the uncertainties of properties obtained using REFPROP. 
It is straightforward for pure fluids, since it is specified in "Fluid Information" for most fluids. However it is not an easy task for mixtures.
For hydrocarbons mixtures I think the GERG monograph is a good place to look for these values. There are also other publications for binary mixtures, which include comparison with experimental data, or the data itself. However since there are not many publications for ternary and more components mixtures it is hard to assess uncertainties. 
Is there any way to get those informations or a rule of thumb? For example if there are experimental data for all binary pairs, the uncertainity of the whole mixture can be aproximated using simple propagation of uncertainites?
What if the mixing parameters used, comes from simillar fluids, and there is not experimental data for a binary pair?
How does using P-R EOS affect the uncertainties?

I would really appreciate your thoughts on this topic.
Best Regards,
Adrian

**Comment by ianhbell** [comment]
Yeah, this is a real can of worms you have opened up here. Even for binaries (same for pure fluids) I would argue that it is still impossible or at least difficult to assess the uncertainty. You can only assess the uncertainty where experimental data exist, and no data means no possibility to estimate the uncertainty. We have tried to generalize the uncertainty estimates by various means, but I don't think any of our results are particularly unambiguous or frankly all that useful.  That is especially true for mixtures. 

A tricky question you didn't ask but is relevant: how do you propose to estimate the uncertainty in unmeasurable properties like entropy or enthalpy?

Another question: what about a 10-component mixture? Or a 100-component mixture?

Another question: are you also considering the uncertainty in the measurands of temperature, pressure (or density), and composition?

I'm not suggesting you shouldn't think about uncertainty, quite the opposite, but it is very challenging to do properly and I would argue it remains an open question in the EOS space.

A couple of papers from my collaborators:

* https://pubs.acs.org/doi/abs/10.1021/acs.jced.9b00689 
* https://www.tandfonline.com/doi/abs/10.1080/00268976.2016.1275856

**Comment by Akwiatkowski-eng** [comment]
Ian, thank you for this valueable answer. I expected that the answer will not be an easy one, but I didn't expect to find a small field on its own! Thank you for linking these papers, I'm sure they will add a value to my research. 

Now I would like to ask, a more specific question. In REFPROP 10 GUI, there is an uncertainity of 5% in heat capacities for R14, using Platzer et. al (1990) equation. (The only one available for R14 in fact). Is this "uncertainty" or "accuracy" (devation between model and measurement, as it was named in the first paper you've linked.)? I've got a copy of the "Thermophysical properties of Refrigerants", and for R14, there is only deviation for isochoric specific heat capacity (data comes from a PhD dissertation from 1961) and the range is -1.5% to 1.5%. Do you have any extra experimental data for specific heat capacity? Where does the value 5% in refprop comes from? I would like to find "accuracy" for gaseous R14 at around 290-310K and 1-2MPa, I've performed some measurements of specific heat capacity with uncertainity around 1%, and the maxium deviation between measured values and ones from refprop is 2%. This 5% from equation is the biggest contributor to my uncertainity analysis and I'm wondering if it can be redeced somehow.  

Thank you,
Adrian

**Comment by nist-aharvey** [comment]
This is not a complete answer, but it is worth noting that care should be taken in using a single number for an uncertainty of a property from REFPROP, because in reality the uncertainty will often be different for different conditions -- particularly between the gas and the liquid.  For example, an uncertainty of 0.5% in density is listed, but if you have a gas at 1 bar pressure, the uncertainty in density will be less because it is approaching the ideal-gas limit which has essentially zero uncertainty.  
Similarly for a gas the value of Cp will be dominated by the ideal-gas heat capacity. [You can see how much of Cp in REFPROP comes from the ideal-gas part by asking for the ideal-gas Cp (Cp0) from the GUI.]  The uncertainty of the ideal-gas heat capacity for a small molecule like CF4 will typically be much smaller than 5%, although it is hard to say how much smaller without looking at what that particular EOS work did for that quantity (one would hope they used a definitive thermochemistry source like the JANAF tables, in which case the uncertainty might be on the order of 1% if they fit it accurately). So if at your conditions the ideal-gas part is 95% of the total value and the ideal-gas part had an uncertainty of 1%, the total uncertainty would certainly be less than 5% (even if the real-gas part was uncertain by, say, 10%).

In our local data archives, I find one source of isochoric heat capacity data for gas-phase R14:
Hwang, Y. -T.; Martin, J. J.
The constant-volume heat capacity of gaseous tetrafluoromethane
AIChE J., 1964, 10, 89-91
There are also some Cp data, but only for the liquid:
Smith, J. H.; Pace, E. L.
The thermodynamic properties of carbon tetrafluoride from 12 K to its boiling point. The significance of the parameter .nu.
J. Phys. Chem., 1969, 73, 4232-6
Cope, A. F.; Reamer, H. H.; Pings, C. J.
The heat capacity of the binary liquid mixture tetrafluoromethane + trifluoromethane near the upper consolute point
Ber. Bunsen-Ges. Phys. Chem., 1972, 76, 318

For the ideal-gas Cp, you can see the JANAF numbers in this document:
https://janaf.nist.gov/pdf/JANAF-FourthEd-1998-Carbon.pdf
See the 40th page of the document (page numbered 588).

**Comment by Akwiatkowski-eng** [comment]
Allan,

Just to check if I get you right. I don't see an option for ideal-gas Cp, but there is an ideal-gas enthalpy. I approximated Cp=dh/dT for points in the JANAF. The approximated value of idea-gas Cp stands for around 99,7% of Cp obtained directly as a property. Average relative difference of REFPROP Cp (approximated ideal gas and property) vs JANAF is 0.3% for temperatures between 250-400K. It can be assumed that this is approximately accuracy of idea-gas part of Cp (assuming JANAF is 100% accurate). Now, the uncertainity of the Cp can be approximated assuming that ideal-gas share is uncertain by before estimated value and the real part is uncertain by a safe value of 10% as you mentioned?  That gives average uncertainity of 1% for range of 280-320K and 1-2MPa. 

I've also checked coefficients in R14.FLD file and they do differ from ones published in "Thermophysical Properties of Refrigerants" by B.Platzer, A.Poltr, G.Maurer. They used 20 coefficients for EOS, in R14.FLD there is 22 a -terms and non-zero values in other terms (excuse me but I have rather basic knowledge of the helmholtz equation).  Similar for the ideal gas heat capacity. The number of coefficient is the same (5) but the values differ.  Does it mean that you've refitted the equations on your own? 

Adrian

**Comment by nist-aharvey** [comment]
Ideal-gas Cp (Cp0) is on the main thermodynamic tab under Options, Properties (right below Cp).
So for example I put in 300 K and 1 MPa and get Cp of 63.362 J/mol/K, with Cp0 of 61.363, while JANAF has 61.288 for Cp0 at 300 K. So the EOS is getting close to the JANAF numbers, which is good.  
Hard to judge the uncertainty of the JANAF numbers; it will be more than zero but probably less than 1% at your temperatures.  The JANAF text talks about a difference from some other calculations by McBride (a NASA work) -- the size of that difference is unfortunately not described in detail, but it seems like the 0.1 mentioned (note calorie units, which would need to be multiplied by 4.184 to give the JANAF units that use joules) probably applies to 300 K and the 0.75 to 1000 K. That difference, maybe combined in quadrature with the degree to which the EOS agrees with JANAF, would provide a decent estimate of the uncertainty of Cp0.

And then as I mentioned you could combine that with a rough and probably conservative estimate of a 10% uncertainty in the increment from Cp0 to Cp.

As for why coefficients in the .FLD file look different than in the 1990 publication (which I don't have handy), that's not really my department. It is possible that the original EOS was in some form with units, so that the coefficients would get changed when the EOS is put into the dimensionless form that REFPROP uses.  But that is just a guess.  The FLD file looks like the EOS was implemented in 1997 by @EricLemmon without any refitting.
